OutSystems/outsystems-ui
# ROU-537: Accessibility improvements identified on team audit for the Overflow Menu

This PR is to add the accessibility improvements for the Overflow Meni

### What was done
- Add ArrowUp and ArrowDown keys functionality to Balloon feature:
   - Added the _currentFocusedElementIndex attribute to store the index of the current focused element.
   - Added the _focusableBalloonElements array attribute to store the element focusable placed inside the balloon.
   - Implemented the _manageFocusInsideBalloon method to manage the focus change inside the balloon when pressing the ArrowUp or ArrowDown keys.
   - Modified the _onkeypressCallback method to trigger the ArrowUp and ArrowDown behavior.
   - Modified the _toggleBalloon method to call the _manageFocusInsideBalloon method when the Balloon is opened and clear the focus attributes when it is closed.
- Add ArrowUp and ArrowDown keys functionality to Overflow Menu pattern:
   - Implemented the _onKeydownCallback method and set it as a keyDown event handler.
   - Modified the open method to include the ArrowUp and ArrowDown behavior

### Test Steps
1. Go to a test page with the Overflow Menu
2. Use the Tab key to focus on the Overflow Menu button
3. Press ArrowDown key
4. Check that the OverflowMenu is opened and the first focusable element is focused
6. Perfom the same test using the ArrowUp
7. Check that the OverflowMenu is opened and the last focusable element is focused
8. Check that it is possible to navigate between the items using the Arrow keys.

### Checklist
